src/fklearn/causal/validation/curves.py

```diff
@@ -202,3 +204,75 @@ def relative_cumulative_gain_curve(df: pd.DataFrame,
                                          min_rows=min_rows, steps=steps, effect_fn=effect_fn)
 
     return np.array([(effect - ate) * (rows / size) for rows, effect in zip(n_rows, cum_effect)])
+
+
+@curry
+def effect_curves(
+    df: pd.DataFrame,
+    treatment: str,
+    outcome: str,
+    prediction: str,
+    min_rows: int = 30,
+    steps: int = 100,
+    effect_fn: EffectFnType = linear_effect,
+) -> pd.DataFrame:
+    """
+     Creates a dataset summarizing the effect curves: cumulative effect, cumulative gain and
+     relative cumulative gain. The dataset also contains two columns referencing the data
+     used to compute the curves at each step: number of samples and fraction of samples used.
+     Moreover one column indicating the cumulative gain for a corresponding random model is
+     also included as a benchmark.
+
+     Parameters
+     ----------
+     df : Pandas' DataFrame
+         A Pandas' DataFrame with target and prediction scores.
+
+     treatment : Strings
+         The name of the treatment column in `df`.
+
+     outcome : Strings
+         The name of the outcome column in `df`.
+
+     prediction : Strings
+         The name of the prediction column in `df`.
+
+     min_rows : Integer
+         Minimum number of observations needed to have a valid result.
+
+     steps : Integer
+         The number of cumulative steps to iterate when accumulating the effect
+
+     effect_fn : function (df: pandas.DataFrame, treatment: str, outcome: str) -> int or Array of int
+         A function that computes the treatment effect given a dataframe, the name of the treatment column and the name
+         of the outcome column.
+
+
+     Returns
+     ----------
+     summary curves dataset: pd.DataFrame
+         The dataset with the results for multiple validation causal curves according to the predictions ordering.
+    """
+
+    size: int = df.shape[0]
+    n_rows: List[int] = list(range(min_rows, size, size // steps)) + [size]
+
+    cum_effect: np.ndarray = cumulative_effect_curve(
```

Comment:
Should't this function be genertic? I mean, why only support the cumm effect and not other curves?

Reply:
hmmm, what other curves? Could you give an example? I was thinking about supporting only the three curves that we have currently: the cumulative effect as shown here and the cumulative gain and the relative cumulative gain that are built on top of the cumulative effect curve.

Reply:
Humm, I get it know. You are already using everything.